src/System.Diagnostics.Process/tests/ProcessThreadTests.cs

```diff
@@ -80,16 +79,21 @@ public void TestStartTimeProperty()
                     return;
                 }
 
-                if (ThreadState.Initialized != thread.ThreadState)
+                if (ThreadState.Initialized != thread.ThreadState && ThreadState.Ready != thread.ThreadState)
                 {
-                    // Time in unix, is measured in jiffies, which is incremented by one for every timer interrupt since the boot time.
-                    // Thus, because there are HZ timer interrupts in a second, there are HZ jiffies in a second. Hence 1\HZ, will
-                    // be the resolution of system timer. The lowest value of HZ on unix is 100, hence the timer resolution is 10 ms.
-                    // On Windows, timer resolution is 15 ms from MSDN DateTime.Now. Hence, allowing error in 15ms [max(10,15)].
-                    long intervalTicks = new TimeSpan(0, 0, 0, 0, 15).Ticks;
-                    long beforeTicks = timeBeforeCreatingProcess.Ticks - intervalTicks;
-                    long afterTicks = DateTime.UtcNow.Ticks + intervalTicks;
-                    Assert.InRange(thread.StartTime.ToUniversalTime().Ticks, beforeTicks, afterTicks);
+                    long starttimeTicks = 0;
+                    starttimeTicks = thread.StartTime.ToUniversalTime().Ticks;
+                    if (starttimeTicks != 0)
```

Comment:
This is to ensure there is a value obtained from the /proc/[pid]/task/[tid]/stat file for starttime.. starttime could be undefined if thread info is created but thread has not started yet.. I searched through the linux docs, and couldn't get clarity on how and when these values are calculated